modules/packages/HDF5_HL.chpl

```diff
@@ -3516,4 +3517,128 @@ module HDF5_HL {
       }
     }
   }
+
+  module HDF5_Chapel {
+    /*
+    // This record could be used to open and close an HDF5 file based on
+    // simple scoping rules.  It is currently not used.
+    record HDF5_file {
+      const filename: string;
+      const file_id: hid_t;
+
+      proc init(name: string, access: c_uint = H5F_ACC_RDONLY) {
+        filename = name;
+        file_id = H5Fopen(name.c_str(), access, H5P_DEFAULT);
+      }
+
+      proc deinit() {
+        H5Fclose(file_id);
+      }
+    }
+    */
+
+    /* Read the dataset named `dsetName` out of all HDF5 files in the
+       directory `dirname` with filenames that begin with `filenameStart`.
+       This will read the files in parallel with one task per locale in the
+       `locs` array.
+     */
+    proc readAllHDF5Files(locs: [] locale, dirname: string, dsetName: string,
+                          filenameStart: string, type eltType, param rank) {
+      use BlockDist, HDF5_WAR;
+
+      var filenames: [1..0] string;
+      for f in findfiles(dirname) {
+        if f.startsWith(dirname + '/' + filenameStart:string) &&
+           f.endsWith(".h5") {
+          filenames.push_back(f);
+        }
+      }
+      const Space = {1..filenames.size};
+      const BlockSpace = Space dmapped Block(Space, locs, 1);
+      var files: [BlockSpace] ArrRec(eltType, rank);
+      forall (f, name) in zip(files, filenames) {
+        var locName = name; // copy this string to be local
+        var file_id = H5Fopen(locName.c_str(), H5F_ACC_RDONLY, H5P_DEFAULT);
+        var dims: [0..#rank] hsize_t;
+        H5LTget_dataset_info_WAR(file_id, dsetName.c_str(), c_ptrTo(dims), nil, nil);
```

Comment:
Hm is there protection against out-of-bounds access to `dims` ? Should we add it?

Reply:
I can add a check that the specified `rank` matches the number of dimensions of the data in the file.